node/node.go

```diff
@@ -359,3 +362,12 @@ func (n *Node) Run(ctx context.Context) {
 func (n *Node) Config() Config {
 	return *n.cfg
 }
+
+func parseVersion(version string, log *utils.ZapLogger) string {
```

Comment:
why do we need this function?

Reply:
The function mainly targets the following scenarios:

1. Versions unobtainable due to missing tags.
2. Version string by default comes with the number of commits and the commit ID (e.g. `v0.11.9-36-g11edfa86`). The function should sanitize it so that it becomes a proper semantic version format.

Reply:
since this logic is mostly p2p related, wdyt about moving this function there and passing `version` string in `p2p.New(...)` ?